README.md

```diff
@@ -122,6 +122,40 @@ Services should instrument the time that it takes them to run and the time they
 
 If an integration test is running against the platform and fails, it's not enough to know just that result; the test should be able to identify exactly which service caused its failure so that fixing it is immediately actionable.
 
+## Be Resilient
+
+Harden services by considering when their dependencies might become unavailable and using circuit breakers to protect themselves against degraded external conditions. Use rate limiting and control rods to build layers of defense.
+
+---
+
+Identify single points of failure and compensate for their unavailability where possible. For example, a service might not be able to make much progress anyway if it's database becomes totally unavailable, but should be able to stay online if its ephemeral Redis gets knocked offline.
+
+### Circuit Breakers
+
+The _circuit breaker_ pattern identifies a certain number of failures from an external service within a configured period of time. During such an event the circuit breaker "opens" so that subsequent requests to that service fail automatically and cheaply without having to wait on an expensive timeout. An external service in this sense might be another service with the platform, or an attached resource like Redis.
```

Comment:
This is pretty implementation specific but I feel often circuit breakers do the wrong thing by opening the circuit even for failures that are "safe". There is a huge difference between a service timing out and one that timely produces error responses.